java/ql/lib/semmle/code/java/dataflow/internal/FlowSummaryImpl.qll

```diff
@@ -859,27 +904,27 @@ module Private {
      * Holds if the value that is being tracked is expected to be stored inside
      * content `c` at `n`.
      */
-    predicate summaryExpectsContent(Node n, ContentSet c) {
+    predicate summaryExpectsContent(SummaryNode n, ContentSet c) {
       exists(SummarizedCallable sc, SummaryNodeState state, SummaryComponentStack stack |
-        n = summaryNode(sc, state) and
+        n = TSummaryInternalNode(sc, state) and
         state.isInputState(sc, stack) and
         stack.head() = SummaryComponent::withContent(c)
       )
     }
 
     pragma[noinline]
     private predicate viableParam(
-      DataFlowCall call, SummarizedCallable sc, ParameterPosition ppos, ParamNode p
+      DataFlowCall call, SummarizedCallable sc, ParameterPosition ppos, SummaryParamNode p
     ) {
       exists(DataFlowCallable c |
         c = inject(sc) and
-        p.isParameterOf(c, ppos) and
+        p = TSummaryParameterNode(sc, ppos) and
         c = viableCallable(call)
```

Comment:
This is morally just
```
call = sc.getACall() and
p = TSummaryParameterNode(sc, ppos)
```
so we might be able to get rid of `c` here? In fact, it feels like we can almost get rid of `inject` altogether?

Reply:
That sounds right, and certainly looks like a step in the right direction in terms of further disentangling things. We just have to be sure that `getACall` is part of the implicit language-specific api.